server/game/gameSystems/DamageSystem.ts

```diff
@@ -153,9 +150,13 @@ export class DamageSystem<TContext extends AbilityContext = AbilityContext, TPro
                 Contract.fail(`Unexpected damage type: ${properties['type']}`);
         }
 
-        // Check if the damage will defeat the card, this can be used by abilities (e.g. Tarfful) to determine if the card will be defeated or not
-        const damageAmount = event.amount ?? event.sourceEventForExcessDamage?.availableExcessDamage ?? 0;
-        event.willDefeat = card.canBeDamaged() && card.damage + damageAmount >= card.getHp();
+        if (card.canBeDamaged()) {
```

Comment:
Since this means the card is an illegal target, we don't want to let it ever be the case that we don't go down this if condition. The Contract asserts are typed to allow narrowing to occur, so we can do:

```typescript
Contract.assertTrue(card.canBeDamaged());

const damageAmount = this.getDamageAmountFromEvent(event);
event.availableExcessDamage = damageAmount - Math.min(damageAmount, card.remainingHp);

// Check if the damage will defeat the card, this can be used by abilities (e.g. Tarfful) to determine if the card will be defeated or not
event.willDefeat = damageAmount >= card.remainingHp;
```

Reply:
Nice! I didn't realise it was possible, that's way better that an if.